docs/docs/tutorial/chapter1/second-page.md

```diff
@@ -48,21 +50,58 @@ const HomePage = () => {
 export default HomePage
 ```
 
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```jsx title="web/src/pages/HomePage/HomePage.tsx"
+import { Link, routes } from '@redwoodjs/router'
+import { MetaTags } from '@redwoodjs/web'
+
+const HomePage = () => {
+  return (
+    <>
+      <MetaTags title="Home" description="Home page" />
+
+      // highlight-start
+      <header>
+        <h1>Redwood Blog</h1>
+        <nav>
+          <ul>
+            <li>
+              <Link to={routes.about()}>About</Link>
+            </li>
+          </ul>
+        </nav>
+      </header>
+      <main>Home</main>
+      // highlight-end
+    </>
+  )
+}
+
+export default HomePage
+```
+
+</TabItem>
+</Tabs>
+
 Let's point out a few things here:
 
 - Redwood loves [Function Components](https://www.robinwieruch.de/react-function-component). We'll make extensive use of [React Hooks](https://reactjs.org/docs/hooks-intro.html) as we go and these are only enabled in function components. You're free to use class components, but we recommend avoiding them unless you need their special capabilities.
 - Redwood's `<Link>` tag, in its most basic usage, takes a single `to` attribute. That `to` attribute calls a [_named route function_](../../router.md#link-and-named-route-functions) in order to generate the correct URL. The function has the same name as the `name` attribute on the `<Route>`:
 
   `<Route path="/about" page={AboutPage} name="about" />`
 
-  If you don't like the name or path that `redwood generate` created for your route, feel free to change it in `Routes.js`! Named routes are awesome because if you ever change the path associated with a route (like going from `/about` to `/about-us`), you need only change it in `Routes.js` and every link using a named route function (`routes.about()`) will still point to the correct place! You can also pass a string to the `to` attribute (`routes.to("/about")`), but now if your path ever changed you would need to find and replace every instance of `/about` to `/about-us`.
+  If you don't like the name or path that `redwood generate` created for your route, feel free to change it in `Routes.js`! Named routes are awesome because if you ever change the path associated with a route (like going from `/about` to `/about-us`), you need only change it in `Routes.js` and every link using a named route function (`routes.about()`) will still point to the correct place! You can also pass a string to the `to` attribute (`to="/about"`), but now if your path ever changed you would need to find and replace every instance of `/about` to `/about-us`.
```

Comment:
Did you want to do the `{js,tsx}` thing here on the couple mentions of `Routes.js`?

Isn't the function call `routes.to("/path")`? Where did `to="/path"` come from? That looks like a prop, but where would that go?

I realized it should have just said "You can also pass a string to the `to` *function*..." instead of *attribute*.

Reply:
I thought you were talking about links. They have a `to` prop. `<Link to={routes.about()} />` or `<Link to="/about" />`
If you were talking about the function that'll navigate directly it's `navigate(routes.about())` or `navigate("/about")`

Reply:
OHHHHHH Yep you're right! Maybe it should say "You can pass a string to the `to` *prop*" then? I used "attribute" meaning HTML attribute but I guess a `<Link>` is really a component and those are props.

Reply:
Done

Reply:
> Did you want to do the {js,tsx} thing here on the couple mentions of Routes.js?

"want"... it's ugly af. 

But especially for TS I think it's helpful because sometimes it's `.ts` and sometimes `.tsx`. I have an idea how to improve this, but want to get the PR merged first. I think I can write a remark plugin and a couple of other things to automatically replace {js,tsx} with the right extension (js or tsx) depending on what code samples the user is viewing.

Reply:
`path/to/file.{js.tsx}` no displays as `path/to/file.js` if you have JavaScript selected for your code blocks